### PR TITLE
refactor(pages): rework airdrop escrow page

### LINKED ISSUE #87: Polish airdrops escrow interfaces
- [x] update `/airdrops/escrow` page
- [x] update `/airdrops/register` page
- [x] update `/airdrops/fund` page

### PR DESCRIPTION
**Pull request by grikomsn**
Fixes #87

## Description

This hefty PR reworks the escrow page including various UI elements regarding escrow notice in airdrops fund and register page.

## Changes

List any technical changes.

- [x] update airdrop prop type
- [x] created various utility fns and components
- [x] reworked airdrops escrow page

## Screenshots

**Initial escrow page**

Screen Shot 2022-03-11 at 04 05 31 AM

**Incomplete escrow**

Screen Shot 2022-03-11 at 04 05 07 AM

**Processing escrow**

Screen Shot 2022-03-11 at 04 06 56 AM

**Intercepted browser close when escrow is processing**

Screen Shot 2022-03-11 at 04 07 15 AM

**Completed escrow**

Screen Shot 2022-03-11 at 04 05 41 AM

## Testing Steps

As a reviewer, what steps should I take to verify this is working correctly?

- use existing cw20 to create airdrop
- proceed to escrow step as usual
- ...

## Links

\-

### CODE REVIEW

File: pages/airdrops/escrow.tsx [hunk 1]
```diff
@@ -81,24 +87,53 @@ const EscrowAirdropPage: NextPage = () => {
...
 
       <hr className="border-white/20" />
 
-      <div className="space-y-8">
+      <Conditional test={airdrop?.escrowStatus === 'processing'}>
+        <div className="flex flex-col flex-grow justify-center items-center p-16 space-y-2 text-center">
+          <CgSpinnerTwoAlt className="animate-spin" size={64} />
+          <h3 className="text-2xl font-bold">Processing Airdrop...</h3>
+          <p className="text-white/50">
+            Grab a cup of coffee, this may take a couple of minutes.
```
Comment (RyanWarner) [inline pages/airdrops/escrow.tsx:131]: ☕

Comment (grikomsn) [reply pages/airdrops/escrow.tsx:131]: Any changes in these lines?

File: components/StackedList.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import clsx from 'clsx'
+import { DetailedHTMLProps, HTMLAttributes, ReactNode } from 'react'
+
+type BaseProps<T = HTMLDListElement> = DetailedHTMLProps<HTMLAttributes<T>, T>
+
+export interface StackedListProps extends BaseProps {
+  children: ReactNode
+}
+
+const StackedList = (props: StackedListProps) => {
```
Comment (RyanWarner) [inline components/StackedList.tsx:10]: if we type this like so, I think we can avoid specifying children here

```suggestion
const StackedList: React.FC<BaseProps> = (props) => {
```

File: components/Alert.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      {AlertIcon && <AlertIcon size={24} />}
+      <div className="flex flex-col flex-grow space-y-2">
+        {children}
+        {/*  */}
+      </div>
+    </div>
+  )
+}
+
+export default Alert
```
Comment (RyanWarner) [inline components/Alert.tsx:46]: I have read that it's better to use named exports for better tree-shaking. Thoughts?

**Review by RyanWarner — CHANGES_REQUESTED** [review]
Great work! A couple minor questions.

File: components/Alert.tsx (hunk 3 above)
Comment (grikomsn) [reply components/Alert.tsx:46]: Yes it does. We could have a separate PR converting conventions and tidy up DX related quirks.

File: components/StackedList.tsx (hunk 2 above)
Comment (grikomsn) [reply components/StackedList.tsx:10]: I forgot one thing after I committed this suggestion. `React.FC` could be problematic both in type definitions and early assumptions. References:

- https://github.com/typescript-cheatsheets/react#section-2-getting-started
- https://fettblog.eu/typescript-react-why-i-dont-use-react-fc

Reverted in 5e74a0a8eddacc7fc7d40f85e4dc6e7cbffaab24.

File: components/Alert.tsx (hunk 3 above)
Comment (RyanWarner) [reply components/Alert.tsx:46]: Separate PR is good 👍 Can you please make an issue for that?

**Review by RyanWarner — APPROVED** [review]

File: components/Alert.tsx (hunk 3 above)
Comment (grikomsn) [reply components/Alert.tsx:46]: https://github.com/CosmosContracts/juno-tools/issues/92

**Review by findolor — APPROVED** [review]
Nice one! 🙌

**Pull request merged**